spike_main/spike.cc

```diff
@@ -319,7 +319,7 @@ int main(int argc, char** argv)
 #ifdef HAVE_BOOST_ASIO
   parser.option('s', 0, 0, [&](const char* s){socket = true;});
 #endif
-  parser.option('p', 0, 1, [&](const char* s){nprocs = atoul_nonzero_safe(s);});
+  parser.option('p', 0, 1, [&](const char* s){cfg.nprocs.set(atoul_nonzero_safe(s));});
```

Comment:
Would it be possible (and desirable) to add `cfg_arg_t::operator=()` so this code could look like a simple assignment to `cfg.nprocs`?

Reply:
I wasn't sure: I'm always a bit averse to stuff that's too magical! But I guess it can't hurt: we can always change it back in future if it's looking confusing. I'll push up a version with both of these changes in a couple of minutes.